### ISSUE TITLE
[Bug]: Taskbar items are not clickable after using Panel colorizer to customize the panel

### ISSUE THREAD

**Issue by HattoriSuzuki** [luisbocanegra/plasma-panel-colorizer#100]
[✓ ] I have searched if this issue hasn't been reported yet.

**Describe the bug**
Taskbar items are not clickable after using Panel colorizer to customize the panel

**To Reproduce**
1. Use Panel colorizer to customize the panel
2. Reboot the system or logout then login again.
3. All task bar items (except the first one) are no longer clickable.
4. See the attachment for screencast
[only-the-first-item-is-clickable.webm](https://github.com/user-attachments/assets/dc1b169d-7071-4534-9bc7-1ce53b51763f)

**Expected behavior**
All items should be clickable.

**Screencast**
[only-the-first-item-is-clickable.webm](https://github.com/user-attachments/assets/dc1b169d-7071-4534-9bc7-1ce53b51763f)

**Project version**
1.0.0

**Desktop (please complete the following information):**
- Operating System: KDE neon 6.2
- KDE Plasma Version: 6.2.1
- KDE Frameworks Version: 6.7.0
- Qt Version: 6.7.2
- Kernel Version: 6.8.0-47-generic (64-bit)
- Graphics Platform: Wayland

**Additional context**
None

**Comment by luisbocanegra** [comment]
Found the reason I think, can you enter edit mode, move the Task manager to the right then re-enable *Native Panel background* from Appearance tab. Then reboot/restart plasma and check if the problem persists?

I replicated your panel layout and that option seems to be what's causing this, also it seems the clicks go through as if there was not panel in there

**Comment by HattoriSuzuki** [comment]
@luisbocanegra You are right, regardless panel layout, disabling the _Native Panel background_ option causes the issue.

**Comment by luisbocanegra** [comment]
Looks like https://bugs.kde.org/show_bug.cgi?id=489086 but us disabling the panel background somehow makes it worse, as if the panel mask (the blur and contrast) was also needed to define the clickable area of the panel.

**Comment by luisbocanegra** [comment]
Do you have that bug too?
This is what the panel look like every boot with that bug:
![image](https://github.com/user-attachments/assets/eafb09d4-f94d-4e54-a903-22f9e47faec4)

**Comment by HattoriSuzuki** [comment]
@luisbocanegra hmm, looks like I'm having the same bug, I desired my bug here https://www.reddit.com/r/kde/comments/1g79i1c/kde_plasma_62_panel_displays_incorrectly_after_a/

**Comment by luisbocanegra** [comment]
Yes it is the same bug, but the panel not being clickable is apparently a different issue related to the mask being turned off as this can be still reproduced by

1. Disabling the native panel background
2. Making the panel size grow by launching a bunch of apps

At this point, the right side of the panel aren't clickable, and toggling the background on makes them clickable again

**Comment by cyberkunju** [comment]
am having the same issue. when ever i restart my pc the dock applications will disappear. and also i cant open that apps. Help me to solve this.!

**Comment by cyberkunju** [comment]
![Screenshot_20241020_144515](https://github.com/user-attachments/assets/1e0f2472-04a0-4d94-8787-5d277d7ab3b8)
this after i setup. and when i restart the pc it will show like this
![Screenshot_20241020_144717](https://github.com/user-attachments/assets/52aa15c0-d8b7-477e-8c21-1ccf808497b7)

What to do? any fix please? i really want this setup. dont know if the problem is because i set panel colourizer for my top bar and also for the dock at the same time.

**Comment by cyberkunju** [comment]
currently mine is solved. i just turn on native background and make the opacity 0. but please fix the bug,

**Comment by luisbocanegra** [comment]
@cyberkunju just to clarify, the panel being shrunk like that is a KDE bug, the panel clickable area not being updated is caused by this widget when the native background is disabled.

I have an idea of a possible workaround for the shrink issue while KDE fixes it, but for the click problem I will have to look at the KDE panel's code to see if it's a bug on their end or something I have to do to avoid it.

Currently the only way to avoid the click problem are

- not disabling the panel background for panels that have **Width** set to **Fit content**
- Or if you are using a custom background to enable *Blur custom background* (the C++ plugin must be installed for this to work)

**Comment by luisbocanegra** [comment]
Added a workaround for both the upstream length bug (https://bugs.kde.org/show_bug.cgi?id=489086) and the click issue (which might also be a KDE bug but will need to investigate this one further to come up with a proper fix)

Please try the branch https://github.com/luisbocanegra/plasma-panel-colorizer/tree/startup-length-and-clickable-area-workaround

```
git clone https://github.com/luisbocanegra/plasma-panel-colorizer.git -b startup-length-and-clickable-area-workaround
cd plasma-panel-colorizer
kpackagetool6 -t Plasma/Applet -u package
```
Restart plasmashell/logout and let me know if it works

**Comment by luisbocanegra** [comment]
[Screencast_20241020_204547.webm](https://github.com/user-attachments/assets/44608bb6-7691-4074-9fb8-453f37a3ea58)

**Comment by HattoriSuzuki** [comment]
Thank you @luisbocanegra, the workaround works great!

**Comment by luisbocanegra** [comment]
> Thank you @luisbocanegra, the workaround works great!
> 

Thanks for the confirmation, if you there are any freezes or plasmashell crashes after this change please let me know. I had a couple yesterday but not sure if it's related to this change or something else on my system.

**Comment by deviaze** [comment]
Thanks!! Also had this same issue where none of my panels were clickable after a restart. Looks like this workaround fixed it... for now.

**Comment by luisbocanegra** [comment]
Merged, thanks everyone for testing
